### ISSUE TITLE
Cannot find rule in rule directory

### ISSUE THREAD

**Issue by angelathewebdev** [eslint/eslint#5523]
**What version of ESLint are you using?**

ESLint 

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

`.eslintrc`

```js
{
  "extends": "airbnb",
  "rules": {
    "no-console-log": 2
  }
}
```

with an additional custom rule `no-console-log.js`

```js
"use strict";
​
//------------------------------------------------------------------------------
// Rule Definition
//------------------------------------------------------------------------------
​
module.exports = function(context) {
​
    return {
​
        "MemberExpression": function(node) {
​
            if (node.object.name === "console" && node.property.name === "log") {
                var blockConsole = true;
​
                if (context.options.length > 0) {
                    var allowedProperties = context.options[0].allow;
                    var passedProperty = node.property.name;
                    var propertyIsAllowed = (allowedProperties.indexOf(passedProperty) > -1);
​
                    if (propertyIsAllowed) {
                        blockConsole = false;
                    }
                }
​
                if (blockConsole) {
                    context.report(node, "Unexpected console statement.");
                }
            }
        }
    };
​
};
​
module.exports.schema = [
    {
        "type": "object",
        "properties": {
            "allow": {
                "type": "array",
                "items": {
                    "type": "string"
                },
                "minItems": 1,
                "uniqueItems": true
            }
        },
        "additionalProperties": false
    }
];
```

**What did you do? Please include the actual source code causing the issue.**

`eslint --rulesdir scripts/rules/ src/js/actions/index.js`

**What did you expect to happen?**

**What actually happened? Please include the actual, raw output from ESLint.**

```sh
  eslint:cli Running on files +0ms
  eslint:cli-engine Loading rules from scripts/rules/ +36ms
  eslint:ignored-paths Looking for ignore file in /Users/xuyan/Documents/platform/platform +5ms
  eslint:ignored-paths Could not find ignore file in cwd +0ms
  eslint:cli-engine Processing /Users/xuyan/Documents/platform/platform/src/js/actions/index.js +3ms
  eslint:cli-engine Linting /Users/xuyan/Documents/platform/platform/src/js/actions/index.js +1ms
  eslint:config Constructing config for /Users/xuyan/Documents/platform/platform/src/js/actions/index.js +0ms
  eslint:config Using .eslintrc and package.json files +1ms
  eslint:config Loading /Users/xuyan/Documents/platform/platform/.eslintrc +2ms
  eslint:config-file Loading config file: /Users/xuyan/Documents/platform/platform/.eslintrc +1ms
Cannot read property 'no-console-log' of undefined
TypeError: Cannot read property 'no-console-log' of undefined
    at /usr/local/lib/node_modules/eslint/lib/config/config-ops.js:176:68
    at Array.forEach (native)
    at deepmerge (/usr/local/lib/node_modules/eslint/lib/config/config-ops.js:175:30)
    at /usr/local/lib/node_modules/eslint/lib/config/config-ops.js:157:34
    at Array.forEach (native)
    at deepmerge (/usr/local/lib/node_modules/eslint/lib/config/config-ops.js:149:30)
    at /usr/local/lib/node_modules/eslint/lib/config/config-ops.js:177:32
    at Array.forEach (native)
    at Object.deepmerge [as merge] (/usr/local/lib/node_modules/eslint/lib/config/config-ops.js:175:30)
    at loadConfigFile (/usr/local/lib/node_modules/eslint/lib/config/config-file.js:223:22)
```

**Comment by afahim** [comment]
@angelathewebdev I couldn't reproduce the issue. Can you mention what version of ESLint you're using? You can find it out using `eslint -v`.

**Comment by alberto** [comment]
Please provide the requested info.
